include/RevMem.h

```diff
@@ -361,6 +375,11 @@ class RevMem {
     return FreeMemSegs;
   }
 
+  ///< RevMem: Get DumpRanges vector
+  std::map< std::string, std::shared_ptr< MemSegment > >& GetDumpRanges() {
+    return DumpRanges;
+  }
```

Comment:
You might want to overload a `const` version too:
```c++
const std::map< std::string, std::shared_ptr< MemSegment > >& GetDumpRanges() const {
    return DumpRanges;
}
```
(`const` added to the return reference type and the `*this` implicit object.)